test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/CSharpToMarkupSwitchTest/GivesAllWhitespaceOnSameLineWithTrailingNewLineToMarkupExclPreceedingNewline.stree.txt

```diff
@@ -1,74 +1,79 @@
-    Markup block - Gen<None> - 14 - (126:6,0)
-        Markup span - Gen<Markup> - [    ] - SpanEditHandler;Accepts:Any - (126:6,0) - Tokens:1
-            SyntaxKind.Whitespace;[    ];
-        Transition span - Gen<None> - [@] - SpanEditHandler;Accepts:None - (130:6,4) - Tokens:1
-            SyntaxKind.Transition;[@];
-        MetaCode span - Gen<None> - [:] - SpanEditHandler;Accepts:Any - (131:6,5) - Tokens:1
-            SyntaxKind.Colon;[:];
-        Markup span - Gen<Markup> - [Hello!LF] - SpanEditHandler;Accepts:None - (132:6,6) - Tokens:3
-            SyntaxKind.Text;[Hello];
-            SyntaxKind.Bang;[!];
-            SyntaxKind.NewLine;[LF];
-    Code span - Gen<Stmt> - [    var biz = boz;LF}] - SpanEditHandler;Accepts:Any - (140:7,0) - Tokens:11
-        SyntaxKind.Whitespace;[    ];
-        SyntaxKind.Identifier;[var];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Identifier;[biz];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Assign;[=];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Identifier;[boz];
-        SyntaxKind.Semicolon;[;];
-        SyntaxKind.NewLine;[LF];
-        SyntaxKind.RightBrace;[}];
+CSharpCodeBlock - [0..161)::161 - [if(foo) {LF    var foo = "After this statement there are 10 spaces";          LF    <p>LF        FooLF        @barLF    </p>LF    @:Hello!LF    var biz = boz;LF}]
+    CSharpTransition - [0..0)::0 - Gen<None> - SpanEditHandler;Accepts:None
+        Transition;[<Missing>];
+    CSharpStatementLiteral - [0..80)::80 - [if(foo) {LF    var foo = "After this statement there are 10 spaces";          LF] - Gen<Stmt> - SpanEditHandler;Accepts:Any
+        Keyword;[if];
+        LeftParenthesis;[(];
+        Identifier;[foo];
+        RightParenthesis;[)];
+        Whitespace;[ ];
+        LeftBrace;[{];
+        NewLine;[LF];
+        Whitespace;[    ];
+        Identifier;[var];
+        Whitespace;[ ];
+        Identifier;[foo];
+        Whitespace;[ ];
+        Assign;[=];
+        Whitespace;[ ];
+        StringLiteral;["After this statement there are 10 spaces"];
+        Semicolon;[;];
+        Whitespace;[          ];
+        NewLine;[LF];
+    MarkupBlock - [80..126)::46
+        MarkupTextLiteral - [80..84)::4 - [    ] - Gen<Markup> - SpanEditHandler;Accepts:Any
+            Whitespace;[    ];
+        MarkupTagBlock - [84..87)::3 - [<p>]
+            MarkupTextLiteral - [84..87)::3 - [<p>] - Gen<Markup> - SpanEditHandler;Accepts:None
+                OpenAngle;[<];
+                Text;[p];
+                CloseAngle;[>];
+        MarkupTextLiteral - [87..102)::15 - [LF        FooLF] - Gen<Markup> - SpanEditHandler;Accepts:Any
+            NewLine;[LF];
+            Whitespace;[        ];
+            Text;[Foo];
+            NewLine;[LF];
+        CSharpCodeBlock - [102..114)::12
+            CSharpStatementLiteral - [102..110)::8 - [        ] - Gen<Stmt> - SpanEditHandler;Accepts:Any
+                Whitespace;[        ];
```

Comment:
By design, `CSharpImplicitExpressionSyntax` cannot have leading whitespace (especially something we want rendered in output). This whitespace will be converted to `MarkupTextLiteral` by `WhitespaceRewriter` after the parsing is done. Once that runs, the cspans for both old and the new tree will be exactly the same.